blocks/url-input/test/button.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import { shallow, mount } from 'enzyme';
+
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+import UrlInputButton from '../button';
+
+describe( 'UrlInputButton', () => {
+	it( 'should has valid class in wrapper tag', () => {
+		const wrapper = shallow( <UrlInputButton /> );
+		expect( wrapper.hasClass( 'blocks-url-input__button' ) ).toBeTruthy();
+	} );
+	it( 'should not have is-active class if url prop not defined', () => {
+		const wrapper = shallow( <UrlInputButton /> );
+		expect( wrapper.find( 'IconButton' ).hasClass( 'is-active' ) ).toBeFalsy();
+	} );
+	it( 'should have is-active class if url prop defined', () => {
+		const wrapper = shallow( <UrlInputButton url="https://example.com" /> );
+		expect( wrapper.find( 'IconButton' ).hasClass( 'is-active' ) ).toBeTruthy();
+	} );
+	it( 'should hidden form for default', () => {
+		const wrapper = shallow( <UrlInputButton /> );
+		expect( wrapper.find( 'form' ).length ).toBe( 0 );
+		expect( wrapper.state().expanded ).toBeFalsy();
+	} );
+	it( 'should visible form if Edit Link button clicked', () => {
+		const wrapper = shallow( <UrlInputButton /> );
+		wrapper.find( 'IconButton.components-toolbar__control' ).simulate( 'click' );
+		expect( wrapper.find( 'form' ).length ).toBe( 1 );
+		expect( wrapper.state().expanded ).toBeTruthy();
+	} );
+	it( 'should call onChange function at once if value changes at once', () => {
+		const onChangeMock = jest.fn();
+		const wrapper = shallow( <UrlInputButton onChange={ onChangeMock } /> );
+		wrapper.setState( { expanded: true } );
+		wrapper.find( { value: '' } ).simulate( 'change' );
+		expect( onChangeMock.mock.calls.length ).toBe( 1 );
+	} );
+	it( 'should call onChange function at twice if value changes at twice', () => {
+		const onChangeMock = jest.fn();
+		const wrapper = shallow( <UrlInputButton onChange={ onChangeMock } /> );
+		wrapper.setState( { expanded: true } );
+		wrapper.find( { value: '' } ).simulate( 'change' );
```

Comment:
Another option is to look for `UrlInput` instead of an element with an empty value:
```js
wrapper.find( 'UrlInput' ).simulate( 'change' );
```

Reply:
Thanks !

But I've try this code, our test doesn't work well..

```
 FAIL  blocks/url-input/test/button.js
  ● UrlInputButton › should call onChange function at once if value changes at once

    Method “simulate” is only meant to be run on a single node. 0 found instead.
      
      at ShallowWrapper.single (node_modules/enzyme/build/ShallowWrapper.js:1614:17)
      at ShallowWrapper.simulate (node_modules/enzyme/build/ShallowWrapper.js:925:21)
      at Object.<anonymous> (blocks/url-input/test/button.js:40:28)
      at process._tickCallback (node.js:438:9)
```

And I've try this code to debug the test.

```
const onChangeMock = jest.fn();
const wrapper = shallow( <UrlInputButton onChange={ onChangeMock } /> );
expect(wrapper.debug()).toBe('')
```

I think `UrlInput` tag is converted as '_class'.

```
<div className=\"blocks-url-input__button\">
      <IconButton icon=\"admin-links\" label=\"Edit Link\" onClick={[Function]} className=\"components-toolbar__control\" />
      <form className=\"blocks-format-toolbar__link-modal\" onSubmit={[Function]}>
        <IconButton className=\"blocks-url-input__back\" icon=\"arrow-left-alt\" label=\"Close\" onClick={[Function]} />
        <_class value=\"\" onChange={[Function]} />
        <IconButton icon=\"editor-break\" label=\"Submit\" type=\"submit\" />
      </form>
    </div>
```

So I'll try to use `data-`attributes to get`UrlInput` component.

reference: https://blog.kentcdodds.com/making-your-ui-tests-resilient-to-change-d37a6ee37269

Reply:
`<_class value=\"\" onChange={[Function]} />` - Yes, somehow it doesn't have display name set. We can use data-test for the time being. I will look into it later.